pkg/domain/config.go

```diff
@@ -0,0 +1,177 @@
+package domain
+
+import (
+	"errors"
+	"fmt"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/util"
+)
+
+type Config struct {
+	Dogus  map[common.SimpleDoguName]CombinedDoguConfig
+	Global GlobalConfig
+}
+
+type CombinedDoguConfig struct {
+	DoguName        common.SimpleDoguName
+	Config          DoguConfig
+	SensitiveConfig SensitiveDoguConfig
+}
+
+type DoguConfig struct {
+	Present map[common.DoguConfigKey]common.DoguConfigValue
+	Absent  []common.DoguConfigKey
+}
+
+type SensitiveDoguConfig struct {
+	Present map[common.SensitiveDoguConfigKey]common.SensitiveDoguConfigValue
+	Absent  []common.SensitiveDoguConfigKey
+}
+
+type GlobalConfig struct {
+	Present map[common.GlobalConfigKey]common.GlobalConfigValue
+	Absent  []common.GlobalConfigKey
+}
+
+func (config Config) validate() error {
+	var errs []error
+	for doguName, doguConfig := range config.Dogus {
+		if doguName != doguConfig.DoguName {
+			errs = append(errs, fmt.Errorf("dogu name %q in map and dogu name %q in value are not equal", doguName, doguConfig.DoguName))
+		}
+
+		errs = append(errs, doguConfig.validate())
+	}
+
+	errs = append(errs, config.Global.validate())
+
+	return errors.Join(errs...)
+}
+
+func (config CombinedDoguConfig) validate() error {
+	var errs []error
+	err := config.Config.validate(config.DoguName)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("config for dogu %q invalid: %w", config.DoguName, err))
+	}
+
+	err = config.SensitiveConfig.validate(config.DoguName)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("sensitive config for dogu %q invalid: %w", config.DoguName, err))
+	}
+
+	return errors.Join(errs...)
+}
+
+func (config DoguConfig) validate(referencedDoguName common.SimpleDoguName) error {
```

Comment:
🚫 **[checkstyle]** <dupl> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>66-105 lines are duplicate of `pkg/domain/config.go:107-145`